templates/IAM/snowflake-synapse-access.yaml

```diff
@@ -0,0 +1,60 @@
+Description: >
+  Setup cross account IAM access for Snowflake.
+  See https://docs.snowflake.com/en/user-guide/data-load-s3-config-storage-integration
+AWSTemplateFormatVersion: 2010-09-09
+Parameters:
+  Stack:
+    Type: String
+    AllowedValues:
+      - dev
+      - prod
+  SnowflakeAccountArn:
+    Type: String
+  SnowflakeAccountExternalId:
+    Type: String
+Resources:
+  SnowflakeServicePolicy:
+    Type: 'AWS::IAM::ManagedPolicy'
+    Properties:
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          -
+            Effect: Allow
+            Action:
+              - s3:GetObject
+              - s3:GetObjectVersion
+            Resource: !Sub "arn:aws:s3:::${Stack}.datawarehouse.sagebase.org/warehouse/*"
+          -
+            Effect: Allow
+            Action:
+                - s3:ListBucket
+                - s3:GetBucketLocation
+            Resource: !Sub "arn:aws:s3:::${Stack}.datawarehouse.sagebase.org"
```

Comment:
to make it more generic i suggest you pass in a bucket ARN and prefix as parameters.
example:

```
Parameters:
  BucketArn:
    Type: String
  BucketPrefix:
    Type: String
```

for the PolicyDocument..
```
      PolicyDocument:
        Version: 2012-10-17
        Statement:
          -
            Effect: Allow
            Action:
              - s3:GetObject
              - s3:GetObjectVersion
            Resource: !Sub "${BucketArn}/${BucketPrefix}/*"
          -
            Effect: Allow
            Action:
                - s3:ListBucket
                - s3:GetBucketLocation
            Resource: !Sub "${BucketArn}"
```

Reply:
Great idea! Renamed since not tied to Synapse anymore. Thx!